### ISSUE TITLE
The program crashes when setting the option "IncludeExpiredSubscriptions" to true

### ISSUE THREAD

**Issue by ManyFaceGoat** [sim0n00ps/OF-DL#597]
## Actual behavior

| Versions tried |
|---|
| `v1.7.76` |
| `v1.7.79` |

When setting this option : 

```json
{
  "IncludeExpiredSubscriptions": true
}
```

The program ends up with these errors : 

```shell
2024-10-10 15:48:44.152 +02:00 [DBG] Full request URL: https://onlyfans.com/api2/v2/subscriptions/subscribes?offset=58&limit=50&type=expired&format=infinite
2024-10-10 15:48:44.230 +02:00 [ERR] Exception caught: Response status code does not indicate success: 429 (Too Many Requests).

StackTrace:    at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at OF_DL.Helpers.APIHelper.BuildHeaderAndExecuteRequests(Dictionary`2 getParams, String endpoint, HttpClient client)
   at OF_DL.Helpers.APIHelper.GetAllSubscriptions(Dictionary`2 getParams, String endpoint, Boolean includeRestricted, IDownloadConfig config)
2024-10-10 15:48:44.235 +02:00 [ERR] Exception caught: Object reference not set to an instance of an object.

StackTrace:    at OF_DL.Program.DownloadAllData(APIHelper m_ApiHelper, Auth Auth, Config Config)
   at OF_DL.Program.Main(String[] args)
```

## Expected behavior

When setting this option : 

```json
{
  "IncludeExpiredSubscriptions": true
}
```

I should be able to keep using the program instead of seeing an Exception 🙂

**Comment by melithine** [comment]
The error says it had a problem with the ignored users list, not expired subscriptions.

**Comment by ManyFaceGoat** [comment]
After investigating with the "Debug" mode, I understand that the programs tries to retrieve the expired subscriptions gradually with the offset increasing (randomly ?).

At some point, the endpoint for the expired subscriptions will respond with : 

```json
{
  "list": [],
  "hasMore": true
}

My guess is that the offset stops increasing on the next requests since the list is empty.
But it keeps requesting forever because `hasMore` is equal to `true` (not really forever since the endpoint ends up responding with a 429 status code).

When I check the behavior on OF, the client keeps requesting with an offset increasing by a STEP of 10, regardless of that.
Until the endpoint responds with :

```json
{
  "list": [],
  "hasMore": false
}

**Comment by ManyFaceGoat** [comment]
> The error says it had a problem with the ignored users list, not expired subscriptions.

I've just removed it from the initial message since it has nothing to do with the issue.

**Comment by sim0n00ps** [comment]
I think the problem here is that the code is checking if the body of the response is equal to `"[]"` which will never be the case because its in a JSON object, so I will change it to check if it contains `"[]"` and hopefully that should fix the problem

**Comment by sim0n00ps** [comment]
This should be fixed in V1.7.80, if you could try and let me know

**Comment by ManyFaceGoat** [comment]
> This should be fixed in V1.7.80, if you could try and let me know

It seems to be fixed indeed.
Thank you for the update 🙂
